packages/compartment-mapper/src/infer-exports.js

```diff
@@ -34,6 +35,14 @@ function* interpretBrowserExports(name, exports) {
  * @yields {[string, string]}
  */
 function* interpretExports(name, exports, tags) {
+  if (isArray(exports)) {
+    yield* interpretExports(
+      name,
+      // Find one that produces non-empty result, discard result and use again
+      exports.find(ex => !interpretExports(name, ex, tags).next().done),
+      tags,
+    );
```

Comment:
Alternative implementation that doesn't perform one unnecessary operation just feels clunky
```js
    let sample,
      alternative,
      i = 0;
    do {
      alternative = interpretExports(name, exports[i], tags);
      sample = alternative.next();
      i++;
    } while (sample.done);

    yield sample.value;
    yield* alternative;
```
This is not a heavy operation, so I decided to go with spawning 2 copies of the successful generator.

Reply:
It took me a moment to read, but I agree it’s better than the alternative you propose. I had imagined something like:

```js
for (const section of exports) {
  const results = [...interpretExports(name, section, tags)];
  if (results.length > 0) {
    yield* results;
    break;
  }
}
```

I’m convinced that your solution is equivalent.

Reply:
I'll replace with this to avoid having to look at `.next().done`. Thanks.